nikola/plugins/compile/pandoc.py

```diff
@@ -49,11 +50,27 @@ def set_site(self, site):
         self.config_dependencies = [str(site.config['PANDOC_OPTIONS'])]
         super().set_site(site)
 
+    def _get_pandoc_options(self, source: str) -> List[str]:
+        config_options: Union[List[str], Dict[str, List[str]]] = self.site.config['PANDOC_OPTIONS']
+        type_ = type(config_options)
+        if type_ is list:
```

Comment:
Prefer `isinstance(config_options, (tuple, list))` and similarly `isinstance(config_options, dict)`

Reply:
Since in this case we know specifically what type it is, checking isinstance is overkill.

Reply:
`isinstance` is better style.

Reply:
All others are fixed except this because it is only a stylistic preference and objectively it is simpler here as we already knew what types are allowed.

Reply:
> `isinstance` is better style.

Not flagged by your linter at least.

Reply:
(Computationally it is much cheaper since we checked the type only once and we don't have to worry about inheritance so that single check is in itself faster than isinstance. But it is not a point here as it is not the bottleneck hence in parentheses.)

Reply:
Well, only slightly cheaper:

```python
In [9]: timeit.timeit('z = type(a); z is list; z is dict', 'a=[]')
Out[9]: 0.16847618900010275

In [10]: timeit.timeit('isinstance(a, list); isinstance(a, dict)', 'a=[]')
Out[10]: 0.22321782099970733
```

And the fact that we don’t have a linter to enforce a specific rule doesn’t mean that we can’t ask contributors to follow them.

Reply:
I still need a reason. Could you explain **why** this is a better style?

I believe we're in the case of superstition (on blinding following PEP8) here. Because I clearly explained why isinstance isn't needed, and the reason I gave is exactly addressing the recommendation made in PEP8.

We're not aiming at checking general types—we knew the allowed types.

And `type_ is list` is hardly not readable as in pseudo-code-like readability.

Reply:
> Well, only slightly cheaper:

I already said that's why it is in parentheses, i.e. not the main argument here.

Reply:
This one specific use case might not be improved by `isinstance`, as I would expect any users to just pass a list or a dict instead of any subtypes. But introducing special exceptions for cases like this one increases the burden on readers of this code and on other maintenance in general. The point of some rules is to make code more consistent and more likely to do the right thing. If we standardize on `isinstance`, we avoid issues caused by usage of `type` in cases where subclasses are allowed/expected/likely/possible. Some code rules might not make much sense in an isolated case, but if you add more rules and more code, it’s better to have simple rules with minimum exceptions to allow for better consistency. And for `isinstance`, the only exception is that we allow `type()` if subclasses shouldn’t pass the check (we have one case like this in Nikola, where we check that something is `datetime.date` but we don’t want `datetime.datetime` to pass that check).

The other reason is that the project you’re contributing to is expecting compliance to this rule and PEP 8 in general (with exceptions for line length, and the weird rules regarding operator spacing — that’s another example of preferring a general one-size-fits-all rule instead of making specific choices in each case).

Reply:
I disagree.

But we've spent too much time discussing this. The time spent discussing is approaching the time spent creating the patch. Feel free to either reject it or accept it. But we should stop wasting time on this small issue.